test/e2e-wi-tests.sh

```diff
@@ -0,0 +1,141 @@
+    gcloud iam service-accounts create ${PUBSUB_SERVICE_ACCOUNT}
+    gcloud projects add-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${PUBSUB_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/pubsub.editor
+    gcloud projects add-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${PUBSUB_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/monitoring.editor
+  fi
+}
+
+# Tear down resources required for Control Plane setup.
+function control_plane_teardown() {
+  # When not running on Prow we need to delete the service accounts and namespaces created
+  if (( ! IS_PROW )); then
+    echo "Tear down ServiceAccount for Control Plane"
+    gcloud iam service-accounts keys delete -q ${CONTROL_PLANE_SERVICE_ACCOUNT_KEY} \
+      --iam-account=${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/pubsub.admin
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/pubsub.editor
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/storage.admin
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/cloudscheduler.admin
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/logging.configWriter
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/logging.privateLogViewer
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/cloudscheduler.admin
+    # Remove iam.serviceAccountAdmin role to the Google service account.
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/iam.serviceAccountAdmin
+    gcloud iam service-accounts delete -q ${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com
+  fi
+}
+
+#create a cluster with Workload Identity enabled.
+initialize $@ --cluster-creation-flag "--workload-pool=\${E2E_PROJECT_ID}.svc.id.goog"
```

Comment:
I simply tried it in the terminal, running echo, only work when there is no `\`, it that expected？

Reply:
Also, for the installation of the cluster, workload identity and secret have some differences. currently, using `ko apply -f config`  will install secret way, you have to apply core/deployment/controller-gke.yaml after that to make workload identity configured. Not sure if I should also include that here?

Reply:
1. Yes, it's expected.
2. I would suggest you add an arg for https://github.com/google/knative-gcp/blob/master/test/lib.sh#L23 and https://github.com/google/knative-gcp/blob/master/test/lib.sh#L31 to determine if we install knative-gcp that has workload identity configured, and pass it as a flag somewhere else.

Reply:
Please replace `${E2E_PROJECT_ID}` with `${PROJECT}` in this line.